src/server/scripts/EasternKingdoms/ArathiBasin/arathi_basin.cpp

```diff
@@ -0,0 +1,788 @@
+            });
+        });
+    }
+
+    void MovementInform(uint32 /*type*/, uint32 pointId) override
+    {
+        switch (pointId)
+        {
+            case 1:
+                me->SetEmoteState(EMOTE_STATE_USE_STANDING);
+                _scheduler.Schedule(6s, [this](TaskContext)
+                {
+                    me->SetEmoteState(EMOTE_ONESHOT_NONE);
+                    // this is a hack
+                    // Flags: 2432696320 (Unknown5, Steering, Unknown10)
+                    // Although, in this path suddenly smooth path is being used
+                    // Flags: 2436890624 (UncompressedPath, Unknown5, Steering, Unknown10)
+                    me->GetMotionMaster()->MoveSmoothPath(2, _pathBack, std::size(_pathBack), true);
+                });
+                break;
+            case 2:
+                me->SetFacingTo(1.8916287f);
+                StartScript();
+                break;
+            default:
+                break;
+        }
+    }
+
+private:
+    TaskScheduler _scheduler;
+
+    Position const _path[15] =
+    {
+        { 856.48090f, 1229.4810f, 14.3900420f },
+        { 857.34180f, 1227.6758f, 13.9702425f },
+        { 858.63380f, 1224.9688f, 13.5952425f },
+        { 858.67365f, 1224.8854f, 13.3668490f },
+        { 858.67380f, 1224.8857f, 13.5952425f },
+        { 859.46680f, 1223.0498f, 13.2202425f },
+        { 860.25977f, 1221.2139f, 12.9702425f },
+        { 860.60420f, 1220.4166f, 12.7002230f },
+        { 861.36456f, 1215.1354f, 11.9964880f },
+        { 859.28990f, 1211.0104f, 11.4966100f },
+        { 851.72740f, 1204.9861f, 9.79202000f },
+        { 847.78990f, 1202.3403f, 9.79787900f },
+        { 844.04865f, 1198.8004f, 10.1364070f },
+        { 838.91490f, 1195.3785f, 10.5601130f },
+        { 836.15625f, 1193.8334f, 11.0199520f }
+    };
+
+    Position const _pathBack[7] =
+    {
+        { 863.21530f, 1207.7517f, 10.196927f },
+        { 865.19100f, 1212.7709f, 10.503690f },
+        { 866.67535f, 1217.4166f, 11.192437f },
+        { 866.73785f, 1221.1337f, 12.000665f },
+        { 865.20490f, 1224.3438f, 12.579007f },
+        { 860.36804f, 1229.3802f, 13.751737f },
+        { 854.49480f, 1233.6702f, 15.472480f }
+    };
+};
+
+struct npc_bg_ab_lumberjack_passive : ScriptedAI
+{
+    npc_bg_ab_lumberjack_passive(Creature* creature) : ScriptedAI(creature) { }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+    void JustAppeared() override
+    {
+        me->SetEmoteState(EMOTE_STATE_SIT_CHAIR_MED);
+        _scheduler.Schedule(5s, 10s, [this](TaskContext context)
+        {
+            me->PlayOneShotAnimKitId(17347); // drink
+            context.Repeat();
+        });
+    }
+
+private:
+    TaskScheduler _scheduler;
+};
+
+struct npc_bg_ab_lumberjack : ScriptedAI
+{
+    npc_bg_ab_lumberjack(Creature* creature) : ScriptedAI(creature) { }
+
+    enum Spells
```

Comment:
Enumarates are normally outside of the struct itself.

Reply:
I kinda like them scoped to the scripts they are using tho, even making some of them now just static const ones
if this is something really desired i will change it, but i would rather do that when everything else is done and its just some code style adjustments if that's okay

Reply:
I believe it is standard in TC not to include it inside the struct. @mdX7 may know better.